floss/utils.py

```diff
@@ -42,6 +43,48 @@ def __call__(self, parser, namespace, values, option_string=None):
         setattr(namespace, self.dest, items)
 
 
+class InstallContextMenu(argparse.Action):
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        super(InstallContextMenu, self).__init__(option_strings, dest, nargs=0, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        # Theoretically, we don't need to check the platform again here,
+        # because non-Windows platforms will not accept the --install-right-click-menu parameter at all.
+        # This judgment is just to make the mypy type check pass.
+        # The same logic applies to `UninstallContextMenu` below.
+        if sys.platform == "win32":
+            import winreg as reg
+
+            menu_name = "Open with FLOSS"
+            menu_command = f'C:\\windows\\system32\\cmd.exe /K "python -m floss ^"%1^""'
+
+            shell_key = reg.OpenKey(reg.HKEY_CURRENT_USER, r"Software\\Classes\\*\\shell", 0, reg.KEY_SET_VALUE)
```

Comment:
on my system, only `Computer\HKEY_CURRENT_USER\SOFTWARE\Classes\*\shellex\` exists, can we create the key or do we need to support multiple registry paths?

Reply:
`Computer\HKEY_CLASSES_ROOT\*\shell\` is another potential path

Reply:
I considered using the `Computer\HKEY_CLASSES_ROOT\*\shell\` path, but it requires administrator privileges to write, and affects other users using the same computer (not all of them have Python and floss installed, and may not have access to the current user's standalone floss.exe). I'll try to create `shell` when it doesn't exist.